lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -268,36 +269,37 @@ void SessionStateEnforcer::m5g_handle_termination_on_timeout(
 void SessionStateEnforcer::m5g_complete_termination(
     SessionMap& session_map, const std::string& imsi,
     const std::string& session_id, SessionUpdate& session_update) {
-  auto map_element = session_map.find(imsi);
-  if (map_element == session_map.end()) {
-    MLOG(MERROR) << "No session found in SessionMap for IMSI " << imsi
-                 << " during terminating session_id " << session_id
-                 << "skipping termination";
-    return;
-  }
 
-  SessionStateUpdateCriteria& session_uc = session_update[imsi][session_id];
-
-  for (auto session_element = map_element->second.begin();
-       session_element != map_element->second.end(); ++session_element) {
-    if ((*session_element)->get_session_id() == session_id) {
-      session_uc.is_session_ended = true;
-      /*Removing session from map*/
-      map_element->second.erase(session_element--);
-      MLOG(MINFO) << "Successfully terminated session " << session_id;
-      /* If it is last session terminated and no session left for this IMSI
-       * remove the imsi as well
-       */
-      if (map_element->second.size() == 0) {
-        session_map.erase(imsi);
-        MLOG(MINFO) << "All sessions terminated for IMSI " << imsi;
-      }
-      return;
-    }
-  }  // end of for loop
-  // reached end of map and no session found.
-  MLOG(MINFO) << "No session found to complete termination"
-              << " session id " << session_id;
+  // If the session cannot be found in session_map, or a new session has
+  // already begun, do nothing.
+  SessionSearchCriteria criteria(imsi, IMSI_AND_SESSION_ID, session_id);
+  auto session_it = session_store_.find_session(session_map, criteria);
+  if (!session_it) {
+    // Session is already deleted, or new session already began, ignore.
+    MLOG(MDEBUG) << "Could not find session for IMSI " << imsi
+                 << " and session ID " << session_id
+                 << ". Skipping termination.";
+  }
+  auto& session    = **session_it;
+  auto& session_uc = session_update[imsi][session_id];
+  if (!session->can_complete_termination(session_uc)) {
+    return;  // error is logged in SessionState's complete_termination

```

Comment:
Why not to add a log message here? The caller of `m5g_complete_termination` will not receive any return message, so it may not be evident to know if this was executed completely or exited prematurely.

Reply:
Not reqd, this is check whether clean up is done or not ?as it is returning from here, the following log messages  will helps us whether it is returned or not.